### PR TITLE
Replace react-moon

### PR DESCRIPTION
**Pull request by cquiroz**
React-moon has been flagged as obsolete lately.
It could be fully removed just using css. This also lets us remove `styled-components`

### CODE REVIEW

File: explore/src/main/scala/explore/targeteditor/ElevationPlotNight.scala [hunk 1]
```diff
@@ -428,15 +428,17 @@ object ElevationPlotNight {
...
-              MoonPhase(
-                phase = moonPhase,
-                size = 20,
-                border = "1px solid black",
-                darkColor = "#303030"
-              ),
-              <.small("%1.0f%%".format(moonIllum * 100))
-            )
+            // Adapted from https://dev.to/thormeier/use-your-i-moon-gination-lets-build-a-moon-phase-visualizer-with-css-and-js-aih
+            <.div(ExploreStyles.MoonSphere)(
```
Comment (hugo-vrijswijk) [inline explore/src/main/scala/explore/targeteditor/ElevationPlotNight.scala:432]: Should we extract this into a separate component similar to the `MoonPhase` component? It doesn't really fit in the responsibility of this component now, I think

File: common/src/main/webapp/sass/solar-system.scss [hunk 2]
```diff
@@ -1,98 +0,0 @@
...
-  }
-
-  to{
-    transform: rotate(360deg);
-  }
-}
-
-/* GRID STYLING */
-
-:root {
```
Comment (hugo-vrijswijk) [inline common/src/main/webapp/sass/solar-system.scss:15]: Have we lost the solar system spinner? 😢

**Review by hugo-vrijswijk — COMMENTED** [review]
Just 1 comment about the component

File: common/src/main/webapp/sass/solar-system.scss (hunk 2 above)
Comment (cquiroz) [reply common/src/main/webapp/sass/solar-system.scss:15]: Yes, it is now on lucuma-ui

File: explore/src/main/scala/explore/targeteditor/ElevationPlotNight.scala (hunk 1 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/ElevationPlotNight.scala:432]: Right, we could make it a component and move it to lucuma-ui

Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/ElevationPlotNight.scala]: I made it a component, I'll move it lucuma-ui later on

**Pull request merged**